src/api/Feed.js

```diff
@@ -394,38 +486,78 @@ class Feed extends Base {
         this.file = file;
         this.errors = [];
         this.errorCallback = onError;
-        const annotationsPromise = shouldShowAnnotations
-            ? this.fetchAnnotations(permissions, shouldShowReplies)
-            : Promise.resolve();
+
+        // Using the UAA File Activities endpoint replaces the need for these calls
+        const annotationsPromise =
+            !shouldUseUAA && shouldShowAnnotations
+                ? this.fetchAnnotations(permissions, shouldShowReplies)
+                : Promise.resolve();
+        const commentsPromise = () => {
```

Comment:
why is the `commentsPromise` declared differently than the others? could it have the same syntax as the others and then doesn't need to be invoked on line ~549?

Reply:
It didn't have a `Promise.resolve()` condition before. It would either fetch ThreadedComments or Comments. In this case, if we have the FF on, we don't want either of the Comment fetches.

Reply:
oh I see now

Reply:
new thought: do these calls: `commentsPromise`, `annotationsPromise`, `tasksPromise`, and `appActivityPromise` need their declarations changed from the original implementation?

from how I understand, these are only used once in the `Promise.all` (~line 559) which is located in the branch when `if (shouldUseUAA == false)` which means these individual functions won't need to check `!shouldUseUAA` right?

Reply:
I think without `shouldUseUAA` in each of the declarations, the calls would be made still, its just that the promises wouldn't be handled. Having `shouldUseUAA` in each prevents making those extra calls. Monoproxy is broken currently so its hard to check, but I think last I checked, without changing the declarations, the calls were being made.